cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/ScheduleTaskRunner.java

```diff
@@ -110,10 +120,21 @@ public ScheduleTaskRunner(Store store, ProgramLifecycleService lifecycleService,
                                       Map<String, String> userArgs) throws Exception {
     ProgramRuntimeService.RuntimeInfo runtimeInfo;
     try {
+      // if the program has a namespace user configured then set that user in the security request context.
+      // See: CDAP-7396
+      String nsPrincipal = namespaceQueryAdmin.get(id.getNamespace()).getConfig().getPrincipal();
+      if (nsPrincipal != null) {
+        SecurityRequestContext.setUserId(new KerberosName(nsPrincipal).getServiceName());
```

Comment:
nsconfig is set only in impersonated ns otherwise it will be null and in that case the normal behavior will happen.